### PR TITLE
TNO-163 Automate Elasticsearch setup

### PR DESCRIPTION
**Pull request by Fosol**
To provide a way to automate setting up Elasticsearch and to run migrations a script has been created that can before updates and rollbacks.

The following basic command will run all migrations and make sure an environment is configured correctly.

```bash
make elastic-update
```

Additional customization is available, refer to the README `./db/elasticsearch/README.md`

There is now a new **nuke** command to reset your local environment.  Use carefullly!

```bash
make nuke
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,90 +38,100 @@ help:
 ##############################################################################
 
 setup: ## Setup local environment for development, generate configuration files.
-	@echo "$(P) Setup local development environment"
+	$(info Setup local environment for development, generate configuration files.)
+	@./tools/scritps/variables.sh
 	@./tools/scripts/gen-env-files.sh
-	@./tools/scripts/gen-dal-db-files.sh
+	@./tools/scripts/gen-maven-files.sh
```
Comment (Fosol) [inline Makefile:44]: Renamed to be more aligned with content of file.

File: Makefile [hunk 2]
```diff
@@ -38,90 +38,100 @@ help:
...
 	@./tools/scripts/gen-keys.sh
 	@mkdir -p ./app/editor/node_modules
 	@mkdir -p ./app/subscriber/node_modules
 
 init: ## Initialize your local environment and start the core solution.
-	@echo "$(P) Initialize your local environment and start the core solution"
+	$(info Initialize your local environment and start the core solution)
```
Comment (Fosol) [inline Makefile:50]: Using `make info` commands instead of `bash echo`

File: Makefile [hunk 3]
```diff
@@ -38,90 +38,100 @@ help:
...
 	@make setup
 	@make up
 	@make db-update
+	@make elastic-update
```
Comment (Fosol) [inline Makefile:54]: On initial setup elasticsearch should now be configured

File: Makefile [hunk 4]
```diff
@@ -38,90 +38,100 @@ help:
...
+	@make stop n=$(n)
 	@make build n=$(n)
 	@make up n=$(n)
 
 remove: ## Remove all containers
-	@echo "$(P) Removing images..."
+	$(info Remove all containers)
 	@docker-compose rm -sv
 
+nuke: ## Stop all containers, delete all containers, volumes, and configuration
```
Comment (Fosol) [inline Makefile:91]: A new way to get your environment all fresh and ready to start over

File: Makefile [hunk 5]
```diff
@@ -135,57 +145,70 @@ kafka-remove: ## Remove the kafka containers
...
 	@docker-compose rm -f -v -s database
 	@docker volume rm -f tno-database-data
 	@make build n=database
 	@make up n=keycloak
 
+##############################################################################
+# Elasticsearch Commands
+##############################################################################
+
+elastic-update: ## Run the elasticsearch migration (n=migration name, r=rollback, i=ignore errors).
```
Comment (Fosol) [inline Makefile:198]: Run the elasticsearch migration script to apply or rollback versions

File: api/editor/Makefile [hunk 6]
```diff
@@ -39,7 +39,7 @@ help:
 
 clean: ## Clean package.
 	@echo "$(P) Clean package"
-	@mvn clean
+	@cd api-editor; mvn clean
```
Comment (Fosol) [inline api/editor/Makefile:42]: Fixed bug

File: api/editor/api-editor/pom.xml [hunk 7]
```diff
@@ -26,6 +26,11 @@
 			<name>Maven Central Staging Repository</name>
 			<url>https://s01.oss.sonatype.org/content/repositories/cabcgovtno-1023</url>
 		</repository> -->
+		<repository>
```
Comment (Fosol) [inline api/editor/api-editor/pom.xml:29]: Required to use SNAPSHOT packages

File: db/elasticsearch/migrations/1.0.0/01-c-index-story.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (Fosol) [inline db/elasticsearch/migrations/1.0.0/01-c-index-story.json:1]: Index configuration mapping for `story` content

File: db/elasticsearch/scripts/migration.sh [hunk 9]
```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
```
Comment (Fosol) [inline db/elasticsearch/scripts/migration.sh:1]: The new elasticsearch migration script

File: tools/scripts/gen-env-files.sh [hunk 10]
```diff
@@ -5,98 +5,6 @@ echo "*************************************"
 echo "Setting up Docker Configuration Files"
 echo "*************************************"
 
-varKeycloak=$(grep -Po 'KEYCLOAK_USER=\K.*$' ./auth/keycloak/.env)
```
Comment (Fosol) [inline tools/scripts/gen-env-files.sh:8]: Moved to `variables.sh` script to keep things clean

File: tools/scripts/gen-keys.sh [hunk 11]
```diff
@@ -9,17 +9,42 @@ echo ""
 echo "Signing keys are required to deploy packages"
```
Comment (Fosol) [inline tools/scripts/gen-keys.sh:9]: Fixed to align with current progress

File: tools/scripts/gen-maven-files.sh [hunk 12]
```diff
@@ -32,6 +32,20 @@ varDbName=$(grep -Po 'POSTGRES_DB=\K.*$' ./db/postgres/.env)
 # TNO Configuration
```
Comment (Fosol) [inline tools/scripts/gen-maven-files.sh:32]: Updated to align with current progress

File: tools/scripts/nuke.sh [hunk 13]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (Fosol) [inline tools/scripts/nuke.sh:1]: Fun

**Review by sturufous — APPROVED** [review]

**Pull request merged**